lib/elixir/lib/calendar.ex

```diff
@@ -361,4 +366,28 @@ defmodule Calendar do
   def get_time_zone_database() do
     Application.get_env(:elixir, :time_zone_database, Calendar.UTCOnlyTimeZoneDatabase)
   end
+
+  @doc false
+  def split_date_time_string(string, caller) do
+    parts = String.split(string, " ", trim: true)
+
+    case List.last(parts) do
+      <<c::utf8, _rest::binary>> = calendar when c in ?A..?Z ->
+        calendar =
+          Module.concat(
+            [calendar]
+            |> Macro.expand(caller)
+          )
```

Comment:
I believe we can figure out the alias like this:

```elixir
module = Module.concat([calendar])

case Keyword.fetch(caller.aliases, module) do
  {:ok, module} -> module
  :error -> module
end
```

that seems to work when I tested it, but not sure if that's totally safe!